api/client/copy.go

```diff
@@ -0,0 +1,113 @@
+package client
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/docker/docker/pkg/archive"
+)
+
+type copyDirection int
+
+const (
+	fromContainer copyDirection = (1 << iota)
+	toContainer
+	acrossContainers = fromContainer | toContainer
+)
+
+// CmdCp handles copying files or directories to/from/across containers.
+func (cli *DockerCli) CmdCp(args ...string) error {
+	cmd := cli.Subcmd("cp", "[SRC_CONTAINER:]SRC_PATH [DST_CONTAINER:]DST_PATH", "Copy files/folders between containers or a local path")
+	if err := cmd.Parse(args); err != nil {
+		return nil
+	}
+
+	if cmd.NArg() != 2 {
+		cmd.Usage()
+		return nil
+	}
+
+	var (
+		srcPath      = cmd.Arg(0)
+		dstPath      = cmd.Arg(1)
+		srcContainer string
+		dstContainer string
+		direction    copyDirection
+	)
+
+	if strings.Contains(srcPath, ":") {
```

Comment:
yea, this isn't a new issue but I think this might make it stand out more.  :-(
One option is to use a different char, or use "::" instead of just ":".
I guess we can leave it as ":" for now and swap it out later since this is an easy one to change once we decide what the magic char(s) should be.  But let's try not to forget about it....